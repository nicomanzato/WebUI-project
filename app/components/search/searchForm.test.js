import React from 'react';
import SearchForm from './searchForm';
import ShallowRenderer from 'react-test-renderer/shallow';

import { mount, ReactWrapper, shallow } from 'enzyme';

describe('search form', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <SearchForm
        animated
        hasSearched={true}
        onTrendInputTextChange={() => {}}
        searchValue={'This is a Test value'}
        onReset={() => {}}
        onSubmit={() => {}}
        onSearchInputTextChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle dispatching its actions', () => {

    const onReset = jest.fn().mockName('mockedFunction');
    const onSubmit = jest.fn().mockName('mockedFunction');

    const wrapper = mount(
      <SearchForm
        animated
        hasSearched={true}
        onTrendInputTextChange={() => {}}
        searchValue={'This is a Test value'}
        onReset={onReset}
        onSubmit={onSubmit}
        onSearchInputTextChange={() => {}}
      />
    );

    const searchTextInput = wrapper
      .find('TextInput')
      .first();

    const searchButton = wrapper
      .find('TouchableHighlight')
      .first();

    const resetButton = wrapper
      .find({name: 'ios-close'})
      .first();

    searchTextInput.props().onChangeText('aSearchKeyword');
    searchButton.props().onPress();
    resetButton.props().onPress();

    expect(onReset).toHaveBeenCalled();
    expect(onSubmit).toHaveBeenCalled();
  });

});
