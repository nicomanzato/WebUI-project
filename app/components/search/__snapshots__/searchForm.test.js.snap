// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search form should render correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFF",
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-evenly",
      "paddingLeft": 5,
      "paddingRight": 5,
      "paddingTop": 17,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "borderColor": "black",
        "borderWidth": 1,
        "flex": 0.65,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "padding": 1,
        "paddingLeft": 5,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#1183ff"
      name="ios-search"
      size={32}
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "flex": 1,
          "fontSize": 17,
          "marginLeft": 3,
          "padding": 5,
        }
      }
      underlineColorAndroid="transparent"
      value="This is a Test value"
    />
    <Icon
      allowFontScaling={false}
      color="#1183ff"
      name="ios-close"
      onPress={[Function]}
      size={32}
      style={
        Object {
          "padding": 5,
        }
      }
    />
  </Component>
  <AnimatedComponent
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1FBFFF",
          "borderRadius": 5,
          "flex": 0.25,
          "height": 40,
          "justifyContent": "center",
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
       Search 
    </AnimatedComponent>
  </AnimatedComponent>
</Component>
`;
